src/wallet/wallet_rpc_server.h

```diff
@@ -129,7 +129,7 @@ namespace tools
       bool on_create_account(const wallet_rpc::COMMAND_RPC_CREATE_ACCOUNT::request& req, wallet_rpc::COMMAND_RPC_CREATE_ACCOUNT::response& res, epee::json_rpc::error& er);
       bool on_label_account(const wallet_rpc::COMMAND_RPC_LABEL_ACCOUNT::request& req, wallet_rpc::COMMAND_RPC_LABEL_ACCOUNT::response& res, epee::json_rpc::error& er);
       bool on_getheight(const wallet_rpc::COMMAND_RPC_GET_HEIGHT::request& req, wallet_rpc::COMMAND_RPC_GET_HEIGHT::response& res, epee::json_rpc::error& er);
-      bool validate_transfer(const std::list<wallet_rpc::transfer_destination>& destinations, const std::string& payment_id, std::vector<cryptonote::tx_destination_entry>& dsts, std::vector<uint8_t>& extra, epee::json_rpc::error& er);
+      bool validate_transfer(const std::list<wallet_rpc::transfer_destination>& destinations, const std::string& payment_id, std::vector<cryptonote::tx_destination_entry>& dsts, std::vector<uint8_t>& extra, bool at_least_one_destination, epee::json_rpc::error& er);
```

Comment:
This change seems to deserve a separate commit?

Reply:
Because it's more involved than just a "if (cond) return;" type line ? It's still a "enforce non empty before dereferencing". Or another reason ?

Reply:
> It's still a "enforce non empty before dereferencing".

But `dsts` being empty is not dereferenced here nor in wallet2::create_transactions_2 (empty `dsts` causes `error::zero_destination`). As such, isn't this change unnecessary?

Reply:
It is in wallet_rpc_server, in on_sweep_all.

Reply:
Ah, I see it now. Thanks for clarifying.